benchmark.py

```diff
@@ -116,22 +148,26 @@
 if args.profile_pyprof:
 	import pyprof
 	pyprof.init()
-	
 if args.profile_cuda:
 	torch.autograd.profiler.emit_nvtx()
 	torch.cuda.profiler.start()
 if args.profile_autograd:
-	autograd_profiler = torch.autograd.profiler.profile(use_cuda = use_cuda, enabled = True, profile_memory = True, record_shapes = True)
+	autograd_profiler = torch.autograd.profiler.profile(
+		use_cuda = use_cuda, enabled = True, profile_memory = True, record_shapes = True
+	)
 	autograd_profiler.__enter__()
 
 print('Starting benchmark for', args.iterations, 'iterations:', 'fwd', '+ bwd' if args.backward else '')
 tic_wall = tictoc()
-times_fwd, times_bwd, fragmentation = torch.zeros(args.iterations), torch.zeros(args.iterations), torch.zeros(args.iterations)
+times_fwd, times_bwd, fragmentation = torch.zeros(args.iterations), torch.zeros(args.iterations), torch.zeros(
+	args.iterations)
+
+loaded_batch = load_batch(batch)
 for i in range(args.iterations):
 	tic = tictoc()
-	y = model(load_batch(batch))
+	y = model(loaded_batch)
 	toc = tictoc()
-	fragmentation[i] = utils.compute_memory_fragmentation()
+	fragmentation[i] = utils.compute_memory_fragmentation_ctypes() if args.onnx else utils.compute_memory_fragmentation()
```

Comment:
please introduce a new array for storing utilisation

Reply:
done